Mathlib/MeasureTheory/Measure/Haar/Basic.lean

```diff
@@ -254,16 +254,18 @@ theorem index_union_eq (K₁ K₂ : Compacts G) {V : Set G} (hV : (interior V).N
       (add_le_add (this K₁.1 <| Subset.trans (subset_union_left _ _) h1s)
         (this K₂.1 <| Subset.trans (subset_union_right _ _) h1s)) _
   rw [← Finset.card_union_of_disjoint, Finset.filter_union_right]
-  exact s.card_filter_le _
+  · exact s.card_filter_le _
   apply Finset.disjoint_filter.mpr
   rintro g₁ _ ⟨g₂, h1g₂, h2g₂⟩ ⟨g₃, h1g₃, h2g₃⟩
   simp only [mem_preimage] at h1g₃ h1g₂
   refine' h.le_bot (_ : g₁⁻¹ ∈ _)
   constructor <;> simp only [Set.mem_inv, Set.mem_mul, exists_exists_and_eq_and, exists_and_left]
-  · refine' ⟨_, h2g₂, (g₁ * g₂)⁻¹, _, _⟩; simp only [inv_inv, h1g₂]
-    simp only [mul_inv_rev, mul_inv_cancel_left]
-  · refine' ⟨_, h2g₃, (g₁ * g₃)⁻¹, _, _⟩; simp only [inv_inv, h1g₃]
-    simp only [mul_inv_rev, mul_inv_cancel_left]
+  · refine' ⟨_, h2g₂, (g₁ * g₂)⁻¹, _, _⟩
```

Comment:
I know you wanted to make minimal changes, but... I think that, like above, `refine` would make the source slightly clearer.

Would you prefer a separate PR with just this clean-up?

Reply:
Done!